app/rpc/namespaces/eth/api.go

```diff
@@ -269,6 +270,40 @@ func (api *PublicEthereumAPI) GasPrice() *hexutil.Big {
 	return api.gasPrice
 }
 
+func (api *PublicEthereumAPI) GasPriceIn3Gears() *rpctypes.GPIn3Gears {
+	monitor := monitor.GetMonitor("eth_gasPriceIn3Gears", api.logger, api.Metrics).OnBegin()
+	defer monitor.OnEnd()
+
+	avgGP := (*big.Int)(api.gasPrice)
+	if appconfig.GetOecConfig().GetDynamicGpMode() != types.MinimalGpMode {
+		price := new(big.Int).Set(app.GlobalGp)
+		if price.Cmp((*big.Int)(api.gasPrice)) == -1 {
+			price.Set((*big.Int)(api.gasPrice))
+		}
+
+		if appconfig.GetOecConfig().GetDynamicGpCoefficient() > 0 {
+			coefficient := big.NewInt(int64(appconfig.GetOecConfig().GetDynamicGpCoefficient()))
+			gpRes := new(big.Int).Mul(price, coefficient)
+			avgGP = gpRes
+		} else {
+			avgGP = price
+		}
+	}
+	// safe low GP = average GP * 0.5, but it will not be less than the minimal GP.
+	safeGp := new(big.Int).Quo(avgGP, big.NewInt(2))
+	if safeGp.Cmp(gasprice.MinPrice) == -1 {
+		safeGp.Set(gasprice.MinPrice)
+	}
+	// fastest GP = average GP * 1.5, but it will not be greater than the max GP.
+	fastestGp := new(big.Int).Add(avgGP, new(big.Int).Quo(avgGP, big.NewInt(2)))
+	if fastestGp.Cmp(gasprice.MaxPrice) == 1 {
```

Comment:
reset avgGP = （min + max）/ 2 again?

Reply:
avgGP is unchanged and does not need to be reset